moby/moby
# Add support for capabilities options in services

**- What I did**
Add support for capabilities options in services

Third step to address moby/moby#25885 , moby/moby#24862 and docker/swarmkit#1030

**- How I did it**
- Increased API version 1.41
- Added Capabilities option
- Bump swarmkit

**- How to verify it**
Create service with command:
```
curl --unix-socket /var/run/docker.sock \
  --header "Content-Type: application/json" \
  --request POST \
  --data '{"Name":"test","TaskTemplate":{"ContainerSpec":{"Image":"ollijanatuinen/capsh","Capabilities":["CAP_NET_RAW","CAP_SYS_CHROOT"]}}}' \
  http://localhost/v1.41/services/create
```
Check log with command: `docker logs <container id>` which should print capabilities from inside of container.

Update capabilities
```
curl --unix-socket /var/run/docker.sock \
  --header "Content-Type: application/json" \
  --request POST \
  --data '{"Name":"test","TaskTemplate":{"ContainerSpec":{"Image":"ollijanatuinen/capsh","Capabilities":["CAP_MKNOD"]}}}}' \
  http://localhost/v1.41/services/test/update?version=7139
```

Reset to default capabilities:
```
curl --unix-socket /var/run/docker.sock \
  --header "Content-Type: application/json" \
  --request POST \
  --data '{"Name":"test","TaskTemplate":{"ContainerSpec":{"Image":"ollijanatuinen/capsh","Capabilities": null}}}}' \
  http://localhost/v1.41/services/test/update?version=7157
```

**- A picture of a cute animal (not mandatory but encouraged)**
![image](https://user-images.githubusercontent.com/6213926/57199503-4b3c7700-6f88-11e9-9a90-778523e1b752.png)
